### DISCUSSION TITLE
[@types/d3] HistogramCommon is expecting only ArrayLike data

### DISCUSSION THREAD

**Discussion by teodorkostov** [DefinitelyTyped/DefinitelyTyped#60686] [Issues with a @types package]
Check the code [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/2d8bb395eacbbdc8431308184ea2702a5c43d6fa/types/d3-array/index.d.ts#L1084).

It should be possible to write:

```typescript
const data: Iterable<MyValue>
const myBins = d3.bins().value((myValue: MyValue) => myValue.v)
myBins(data) // type error that data is not ArrayLike
```
The `data` argument is of the `ArrayLike<T>` type. Such a definition is not correct since iterable objects are valid arguments.

[a comment by an automated account is not shown]

**Comment by Fil** [comment]
Correct. Since [d3@6](https://github.com/d3/d3/releases/tag/v6.0.0) most d3 methods also accept iterables where they used to accept only arrays.
